sdk/include/opentelemetry/sdk/trace/sampler.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include "opentelemetry/common/attribute_value.h"
+#include "opentelemetry/nostd/span.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/sdk/common/atomic_shared_ptr.h"
+#include "opentelemetry/trace/span.h"
+#include "opentelemetry/trace/trace_id.h"
+#include "opentelemetry/trace/tracer.h"
+#include "opentelemetry/version.h"
+
+#include <map>
+#include <memory>
+#include <string>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace trace
+{
+namespace trace_api = opentelemetry::trace;
+/**
+ * NOT_RECORD - IsRecording() == false,
+ *    span will not be recorded and all events and attributes will be dropped.
+ * RECORD - IsRecording() == true, but Sampled flag MUST NOT be set.
+ * RECORD_AND_SAMPLED - IsRecording() == true AND Sampled flag` MUST be set.
+ */
+enum class Decision
+{
+  NOT_RECORD,
+  RECORD,
+  RECORD_AND_SAMPLE
+};
+
+/**
+ * A placeholder for common::AttributeKeyValue
+ * A key/value pair that can be used to set attributes.
+ */
+struct AttributeKeyValue {
+    nostd::string_view key;
+    common::AttributeValue value;
+};
+
+struct SamplingResult
+{
+  Decision decision;
+  // TODO: Change AttributeKeyValue to common::AttributeKeyValue
+  std::unique_ptr<nostd::span<AttributeKeyValue>> attributes;
```

Comment:
We can use a `std::unique_ptr<std::map<std::string, openelemetry::common::AttributeValue>>` here. And let's document that it can be `nullptr`.

Reply:
I want to double check my understanding of the `SamplingResult`. Since `ShouldSample` returns `SamplingResult`, we allocate a new map for `attributes` and pass the ownership to the caller. But in this case how do we make sure "The list of attributes returned by SamplingResult MUST be immutable" as per the spec https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#sampler?